src/draw/sw/lv_draw_sw_img.c

```diff
@@ -284,94 +295,94 @@ static void img_draw_core(lv_draw_unit_t * draw_unit, const lv_draw_image_dsc_t
             blend_dsc.src_stride = blend_w * lv_color_format_get_size(cf_final);
         }
 
-        while(blend_area.y1 <= y_last) {
-            /*Apply transformations if any or separate the channels*/
-            lv_area_t relative_area;
-            lv_area_copy(&relative_area, &blend_area);
-            lv_area_move(&relative_area, -img_coords->x1, -img_coords->y1);
-            if(transformed) {
-                lv_draw_sw_transform(draw_unit, &relative_area, src_buf, src_w, src_h, img_stride,
-                                     draw_dsc, sup, cf, tmp_buf);
-            }
-            else if(draw_dsc->recolor_opa >= LV_OPA_MIN) {
-                int32_t h = lv_area_get_height(&relative_area);
-                if(cf_final == LV_COLOR_FORMAT_RGB565A8) {
-                    uint32_t stride_px = img_stride / 2;
-                    const uint8_t * rgb_src_buf = src_buf + stride_px * 2 * relative_area.y1 + relative_area.x1 * 2;
-                    const uint8_t * a_src_buf = src_buf + stride_px * 2 * src_h + stride_px * relative_area.y1 +
-                                                relative_area.x1;
-                    uint8_t * rgb_dest_buf = tmp_buf;
-                    uint8_t * a_dest_buf = (uint8_t *)blend_dsc.mask_buf;
-                    int32_t i;
-                    for(i = 0; i < h; i++) {
-                        lv_memcpy(rgb_dest_buf, rgb_src_buf, blend_w * 2);
-                        lv_memcpy(a_dest_buf, a_src_buf, blend_w);
-                        rgb_src_buf += stride_px * 2;
-                        a_src_buf += stride_px;
-                        rgb_dest_buf +=  blend_w * 2;
-                        a_dest_buf += blend_w;
-                    }
+        if(LV_RESULT_INVALID == LV_DRAW_SW_IMAGE(&draw_dsc, &blend_dsc)) while(blend_area.y1 <= y_last) {
```

Comment:
Same here, return early to reduce nesting level.

Reply:
Sorry, I just noticed that there is a `lv_free(tmp_buf);` after the  `while(blend_area.y1 <= y_last) {...}`.

So we cannot make this change.

Reply:
@kisvegabor any thoughts?

Reply:
The size of `tmp_buf` is limited to a few image lines. I'm not sure Arm2D needs this buffer. If it's really not required, we can allocate it later if `LV_DRAW_SW_IMAGE` failed.

Reply:
You are right! I will update the code accordingly.

Reply:
Done.